consumer.go

```diff
@@ -392,6 +394,31 @@ func (child *partitionConsumer) HighWaterMarkOffset() int64 {
 	return atomic.LoadInt64(&child.highWaterMarkOffset)
 }
 
+func (child *partitionConsumer) responseFeeder() {
+	for response := range child.feeder {
+		switch err := child.handleResponse(response); err {
+		case nil:
+			child.broker.acks <- nil
+		case ErrOffsetOutOfRange:
+			// there's no point in retrying this it will just fail the same way again
+			// so shut it down and force the user to choose what to do
+			child.AsyncClose()
+			fallthrough
+		default:
+			child.sendError(err)
+			fallthrough
+		case ErrUnknownTopicOrPartition, ErrNotLeaderForPartition, ErrLeaderNotAvailable:
+			// these three are not fatal errors, but do require redispatching
+			Logger.Printf("consumer/%s/%d abandoned broker %d because %s\n", child.topic, child.partition, child.broker.broker.ID(), err)
+			child.broker.acks <- child
+			child.trigger <- none{}
+		}
```

Comment:
It's just way harder to parse in my mind, especially with the `default` case not be the last one. I'd prefer a bunch of `if` statements.